apps/agent/app/src/lib/token-utils.js

```diff
@@ -75,3 +97,46 @@ export async function getTokenName(app, address) {
 
   return tokenName || null
 }
+
+export function getPresetTokens(networkType) {
+  return PRESET_TOKENS.get(networkType) || [ETHER_TOKEN_FAKE_ADDRESS]
+}
+
+export function findTransfersFromReceipt(receipt) {
+  const { logs = [] } = receipt
+  const transferLogs = logs.filter(
+    ({ topics = [] }) => topics[0] === TOKEN_TRANSFER_EVENT_SIGNATURE
+  )
+  try {
+    return transferLogs.map(({ address, data, topics }) => ({
+      token: address,
+      returnData: abi.decodeLog(TOKEN_TRANSFER_EVENT_INPUTS, data, topics),
+    }))
+  } catch (_) {
+    return []
+  }
```

Comment:
Hmm, I think this might lead to inconsistent states on multiple transactions I.e: having 3 transactions but missing 1 in the middle due to the failure of the `abi.decodeLog`, which leads to confusion if a user inspects this particular multiple transaction.